### PR TITLE
[Issue #370] View Recommended Words

### PR DESCRIPTION
**Pull request by salmafg**
#370

@m0hamed 
@nohahesham 
@AmrARaouf 
@Fattouh92 

Please review.

### CODE REVIEW

**Comment by abulazm22** [comment]
alt and opt should be on the place where they are called

File: arability/app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -63,54 +88,34 @@ def index
...
-      respond_to do |format|
-        if @project.save
-          format.html { redirect_to "/developers/projects",
-            notice: I18n.t('views.project.flash_messages.project_was_successfully_created') }
+    @project = Project.new(params[:project].except(:category))
+    @project = Project.createproject(params[:project],current_developer.id)
+    respond_to do |format|
+      if @project.save
+        format.html { redirect_to "/developers/projects",
+          flash: { :success => I18n.t('views.project.flash_messages.project_was_successfully_created') } }
```
Comment (nohahesham) [inline arability/app/controllers/projects_controller.rb]: dont use =>

File: arability/app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -180,32 +193,91 @@ def update
...
+  def get_recommended_words
+    @project = Project.find(params[:project_id])
+    if params[:synonym] != nil
+      params[:synonym].each do |key, syn|
+        @word_id = Keyword.find((key.to_i)).id
+        @synonym_id = syn.to_i
+          # if MySubscription.get_permissions(current_developer.id, 2)
+          @added_word = PreferedSynonym.add_keyword_and_synonym_to_project(
+            @synonym_id, @word_id, @project.id)
+        # end
```
Comment (nohahesham) [inline arability/app/controllers/projects_controller.rb]: remove commented lines

File: arability/app/controllers/projects_controller.rb [hunk 3]
```diff
@@ -51,10 +51,35 @@ def index
...
+  # Params:
+  #   None
+  # Success:
+  #   An empty project will be instantiated
+  # Failure:
+  #   If not signed in he will be redirected to the sign in page.
+  #   If he's exceeded the max number for projects, he will be redirected to the subscription model page.
+  def new
+    # if current_developer.my_subscription.get_projects_limit
+      @project = Project.new
```
Comment (nohahesham) [inline arability/app/controllers/projects_controller.rb:128]: remove commented line

**Comment by abulazm22** [comment]
change ref to opt.
make the last two opt smaller.
state redirects
remove check(remove box from if

**Comment by salmafg** [comment]
@Fattouh92

File: arability/app/models/project.rb [hunk 4]
```diff
@@ -3,72 +3,101 @@ class Project < ActiveRecord::Base
...
 # Success:
-#   Returns a string of category names.
+#   Returns an array of relevant keywords.
 # Failure:
 #   None
-def self.printarray(array)
-  t = array.map {|item| item.name}
-  t = t.join(", ")
-  return t
+def self.filterkeywords(p_id,cat_id)
```
Comment (AmrARaouf) [inline arability/app/models/project.rb]: please rename the method to filter_keywords as per code conventions

File: arability/app/models/project.rb [hunk 5]
```diff
@@ -3,72 +3,101 @@ class Project < ActiveRecord::Base
...
 end
 
 # Author:
 #   Salma Farag
 # Description:
-#   A method that takes an array of categories, maps their names into an array and joins
-#   the array using commas.# Params:
-#   Array of categories.
+#   A method that takes a category id of a project as a parameter and searches the database
+#   for keywords that have the same category, then adds them to an array of keywords. A keyword
```
Comment (AmrARaouf) [inline arability/app/models/project.rb:80]: that has*

File: arability/app/models/project.rb [hunk 6]
```diff
@@ -3,72 +3,101 @@ class Project < ActiveRecord::Base
...
+#   is dsregarded if it does not have any synonyms.
```
Comment (AmrARaouf) [inline arability/app/models/project.rb]: disregarded*

File: arability/app/models/project.rb (hunk 5 above)
Comment (salmafg) [inline arability/app/models/project.rb:80]: Have not has

**Comment by salmafg** [comment]
@AmrARaouf Check now

File: arability/app/controllers/projects_controller.rb [hunk 7]
```diff
@@ -180,32 +193,91 @@ def update
...
-end  
-  
 
 
+  # Author:
+  #   Salma Farag
+  # Description:
+  #   After checking that the user is not a free trial limit of words,
+  #   the method finds the project by its id and then calls the method
+  #   filterkeywords in Project tha filters the relevant keywords.
```
Comment (AmrARaouf) [inline arability/app/controllers/projects_controller.rb]: the\* and change the method name accordingly

**Comment by salmafg** [comment]
@AmrARaouf

File: arability/app/models/project.rb (hunk 5 above)
Comment (AmrARaouf) [inline arability/app/models/project.rb:80]: sorry my bad

File: arability/spec/controllers/projects_controller_spec.rb [hunk 8]
```diff
@@ -1,8 +1,17 @@
 #encoding: UTF-8
 require "spec_helper"
 require "request_helpers"
+<<<<<<< HEAD
```
Comment (AmrARaouf) [inline arability/spec/controllers/projects_controller_spec.rb]: Salma again?

**Comment by AmrARaouf** [comment]
doc verified

**Comment by salmafg** [comment]
@AmrARaouf

File: arability/spec/models/my_subscription_spec.rb [hunk 9]
```diff
@@ -1,3 +1,4 @@
```
Comment (AmrARaouf) [inline arability/spec/models/my_subscription_spec.rb]: please check all the tests here

**Comment by AmrARaouf** [comment]
test verified

**Comment by AmrARaouf** [comment]
please solve merge conflicts so I can merge you

**Pull request merged**